woodwork/schema.py

```diff
@@ -219,8 +221,15 @@ def _filter_cols(self, include):
                     continue
                 elif selector in tags_in_schema:
                     tags_used.add(selector)
+                elif col_names and selector in self.columns:
+                    cols_to_include.add(selector)
             else:
-                raise TypeError(f"Invalid selector used in include: {selector} must be either a string or LogicalType")
+                # Non-string column names are relevant if using col_names
+                if col_names and selector in self.columns:
+                    cols_to_include.add(selector)
+                else:
+                    raise TypeError(f"Invalid selector used in include: {selector} must be a "
+                                    "string, LogicalType, or valid column name")
```

Comment:
Want to highlight that `_filter_cols` was previously not allowing non-string column names to be filtered for. Changed that as well as the error language here.